azure-kusto-data/azure/kusto/data/_kusto_client.py

```diff
@@ -223,6 +192,116 @@ def _get_table(self, table_id):
     }
 
 
+class _KustoResponseDataSet(six.Iterator):
+    """Wrapper for response."""
+
+    def __init__(self, json_response, is_v2):
+        self.is_v2 = is_v2
+        self._index = 0
+        if self.is_v2:
+            deta_set_header = list(
+                filter(lambda x: x["FrameType"] == "DataSetHeader", json_response)
+            )[0]
+            deta_set_completion = list(
+                filter(lambda x: x["FrameType"] == "DataSetCompletion", json_response)
+            )[0]
+            json_response.remove(deta_set_header)
+            json_response.remove(deta_set_completion)
+            self.deta_set_header = _DataSetHeader(deta_set_header)
+            self.deta_set_completion = _DataSetCompletion(deta_set_completion)
+            self.tables = [_KustoResultTable(json) for json in json_response]
+            self.tables_count = len(self.tables)
+        else:
+            if len(json_response["Tables"]) == 1:
+                self.tables = [_KustoResultTable(json_response["Tables"][0])]
+                self.tables_count = 1
+            else:
+                tables = json_response["Tables"][:-1]
+                toc = _KustoResultTable(json_response["Tables"][-1])
+                self.tables = []
+                for i in range(0, len(tables)):
+                    table = tables[i]
+                    table["TableName"] = toc
+                    table["TableName"] = toc[i]["Name"]
+                    table["TableId"] = toc[i]["Id"]
+                    table["TableKind"] = toc[i]["Kind"]
+                    self.tables.append(_KustoResultTable(table))
+
+            self.tables_count = len(self.tables)
+
+    @property
+    def primary_results(self):
+        if self.tables_count == 1:
+            return self.tables[0]
+        primary = list(filter(lambda x: x.table_kind == "PrimaryResult", self.tables))
+        if len(primary) == 1:
+            return primary[0]
+        return primary
```

Comment:
It can be more than one table with "PrimaryResult" as table_kind ?
Don't quite understand the separation of when len(primary) equals 1 and when not.

Reply:
Yes. Think about the following query:
T | fork
( take 1 )
( take 1 )